src/table_builder_test.cc

```diff
@@ -318,6 +344,121 @@ TEST_F(TableBuilderTest, NumEntries) {
   ASSERT_OK(table_builder->Finish());
 }
 
+TEST_F(TableBuilderTest, TargeSize) {
+  cf_options_.blob_file_target_size = kTargetBlobFileSize;
+  table_factory_.reset(new TitanTableFactory(
+      db_options_, cf_options_, blob_manager_, &mutex_, vset_.get(), nullptr));
+  std::unique_ptr<WritableFileWriter> base_file;
+  NewBaseFileWriter(&base_file);
+  std::unique_ptr<TableBuilder> table_builder;
+  NewTableBuilder(base_file.get(), &table_builder);
+  const int n = 255;
+  for (unsigned char i = 0; i < n; i++) {
+    std::string key(1, i);
+    InternalKey ikey(key, 1, kTypeValue);
+    std::string value(kMinBlobSize, i);
+    table_builder->Add(ikey.Encode(), value);
+  }
+  ASSERT_OK(table_builder->Finish());
+  uint64_t file_size = 0;
+  ASSERT_OK(env_->GetFileSize(blob_name_, &file_size));
+  ASSERT_TRUE(file_size < kTargetBlobFileSize + 1 + kMinBlobSize);
+  ASSERT_TRUE(file_size > kTargetBlobFileSize - 1 - kMinBlobSize);
+}
+
+TEST_F(TableBuilderTest, LevelMerge) {
+  cf_options_.blob_file_target_size = kTargetBlobFileSize;
+  cf_options_.level_merge = true;
+  table_factory_.reset(new TitanTableFactory(
+      db_options_, cf_options_, blob_manager_, &mutex_, vset_.get(), nullptr));
+  std::unique_ptr<WritableFileWriter> base_file;
+  NewBaseFileWriter(&base_file);
+  std::unique_ptr<TableBuilder> table_builder;
+  NewTableBuilder(base_file.get(), &table_builder, 0 /* target_level */);
+
+  // generate a level 0 sst with blob file
+  const int n = 255;
+  for (unsigned char i = 0; i < n; i++) {
+    std::string key(1, i);
+    InternalKey ikey(key, 1, kTypeValue);
+    std::string value(kMinBlobSize, i);
+    table_builder->Add(ikey.Encode(), value);
+  }
+  ASSERT_OK(table_builder->Finish());
+  ASSERT_OK(base_file->Sync(true));
+  ASSERT_OK(base_file->Close());
+
+  std::unique_ptr<TableReader> base_reader;
+  NewTableReader(&base_reader);
+
+  std::string first_base_name = base_name_;
+  base_name_ = base_name_ + "second";
+  NewBaseFileWriter(&base_file);
```

Comment:
use another variable to avoid confusing the second base file with the first one?

Reply:
Ok. It's easier to reuse code using base_name_.